### PR TITLE
[Datafactory] Add Support for Managed Private Endpoints and Virtual Networks Resources

### PR DESCRIPTION
**Pull request by drosales57**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/drosales57/azure-rest-api-specs/darosale%2Fdev-datafactory-Microsoft.DataFactory-2018-06-01...master/).

### Contribution checklist:
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for AME onboarding.
- [x] [Validation tools](https://aka.ms/swaggertools) have been run on swagger spec(s) updated and errors have all been fixed with this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, [please view the FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

- [x] Service team MUST add the "**WaitForARMFeedback**" label if the management plane API changes fall into one of the below categories.
*	adding/removing APIs.
* adding/removing properties.
* adding/removing API-version.
* adding a new service in Azure.

*Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.*

- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them.

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by drosales57** [comment]
Hi @qianwens my changes are ready for review. Please take a look, thanks

**Review by qianwens — APPROVED** [review]

**Comment by majastrz** [comment]
As per https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md additions of new endpoints fall under the umbrella of evolutionary changes and therefore require an API version bump.

**Comment by drosales57** [comment]
Hi @majastrz I am not adding a new endpoint. I am adding CRUD APIs for two new subresources for Azure Data Factory. The resource names are managed Private Endpoint and managed Virtual Network.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json [hunk 1]
```diff
@@ -3616,6 +3616,379 @@
...
+        "responses": {
+          "200": {
+            "description": "OK.",
+            "schema": {
+              "$ref": "#/definitions/ManagedPrivateEndpointResource"
+            }
+          },
+          "304": {
+            "description": "Not modified."
+          },
```
Comment (majastrz) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json:3937]: GETs should only return 200 OK or an error.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json [hunk 2]
```diff
@@ -3616,6 +3616,379 @@
...
+        "responses": {
+          "200": {
+            "description": "OK.",
+            "schema": {
+              "$ref": "#/definitions/ManagedVirtualNetworkResource"
+            }
+          },
+          "304": {
+            "description": "Not modified."
+          },
```
Comment (majastrz) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json:3769]: GETs should only return 200 OK or an error.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/examples/ManagedPrivateEndpoints_Create.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  "parameters": {
+    "subscriptionId": "12345678-1234-1234-1234-12345678abc",
+    "resourceGroupName": "exampleResourceGroup",
+    "factoryName": "exampleFactoryName",
+    "managedVirtualNetworkName": "exampleManagedVirtualNetworkName",
+    "managedPrivateEndpointName": "exampleManagedPrivateEndpointName",
+    "ifMatch": null,
+    "managedPrivateEndpoint": {
+      "properties": {
+        "description": "A managed private endpoint",
```
Comment (majastrz) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/examples/ManagedPrivateEndpoints_Create.json:11]: >description [](start = 9, length = 11)

Why is description not being returned in PUT response?

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/examples/ManagedVirtualNetworks_Create.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
+{
+  "parameters": {
+    "subscriptionId": "12345678-1234-1234-1234-12345678abc",
+    "resourceGroupName": "exampleResourceGroup",
+    "factoryName": "exampleFactoryName",
+    "managedVirtualNetworkName": "exampleManagedVirtualNetworkName",
+    "ifMatch": null,
+    "managedVirtualNetwork": {
+      "properties": {
+        "description": "A managed Virtual network"
```
Comment (majastrz) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/examples/ManagedVirtualNetworks_Create.json:10]: >description [](start = 9, length = 11)

Same question here

Comment (drosales57) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/examples/ManagedVirtualNetworks_Create.json]: Removed the description as it is currently not supported.

Comment (drosales57) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/examples/ManagedVirtualNetworks_Create.json]: thanks

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/examples/ManagedPrivateEndpoints_Create.json (hunk 3 above)
Comment (drosales57) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/examples/ManagedPrivateEndpoints_Create.json]: Removed the description. It is not supported. Thanks for catching the issue

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json (hunk 1 above)
Comment (drosales57) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json]: Fixed, thanks

**Review by majastrz — APPROVED** [review]
Signed off from ARM side.

**Comment by drosales57** [comment]
@qianwens can please help merge this PR?

**Pull request merged**